pkg/credentials/sts_tls_identity.go

```diff
@@ -33,12 +33,6 @@ import (
 // livetime.
 type CertificateIdentityOption func(*STSCertificateIdentity)
 
-// CertificateIdentityWithTransport returns a CertificateIdentityOption that
-// customizes the STSCertificateIdentity with the given http.RoundTripper.
-func CertificateIdentityWithTransport(t http.RoundTripper) CertificateIdentityOption {
-	return CertificateIdentityOption(func(i *STSCertificateIdentity) { i.Client.Transport = t })
-}
-
```

Comment:
Specifying a custom transport doesn't make much sense. If we do want to be able to use custom transports for STS, then I need to make a generic fix for all STS options.

Reply:
I have brought back the `http.Client` fields in all providers. Instead of being required, they are now optional. When set, they override the `http.Client` in `CredsContext`.